src/core/lender/tranche/restricted_operator.sol

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2020 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "./operator.sol";
+
+// RestrictedOperator restricts the allowance of users
+contract RestrictedOperator is Operator {
+    mapping (address => uint) maxCurrency;  // uint(-1) unlimited access by convention
+    mapping (address => uint) maxToken;     // uint(-1) unlimited access by convention
+
+    constructor(address tranche_, address assessor_)
+    Operator(tranche_, assessor_) public {}
+
+    function approve(address usr, uint maxToken_, uint maxCurrency_) public auth {
+        if(investors[usr] == 0) {
```

Comment:
I actually think we could get rid of investors in this operator because if an allowance is set that's permission enough. I'd say let's drop the investor_auth modifier on the invest & redeem method and just check that they have an allowance.

I don't know if it make sense to even inherit from the operator or just create a completely new one ( likely not enough code is shared).